core/templates/dev/head/forms/formBuilder.js

```diff
@@ -906,10 +913,16 @@ oppia.filter('isFloat', [function() {
       return undefined;
     }
 
-    if (viewValue !== '' && viewValue !== '-' && FLOAT_REGEXP.test(viewValue)) {
-      return parseFloat(viewValue.replace(',', '.'));
-    }
-    return undefined;
+    if (viewValue !== '' && FLOAT_REGEXP.test(viewValue)) {
+      if (viewValue.slice(-1) === '%'){
+
+        // This is a percentage, so the input needs to be divided by 100.
+        return parseFloat(viewValue.replace(',', '.')) / 100;
```

Comment:
Technically this should be viewValue.substring(0, viewValue.length - 1).replace(',', '.'), right?